src/DataWrangling/ECCO/ECCO_metadata.jl

```diff
@@ -237,10 +241,48 @@ function download_dataset(metadata::ECCOMetadata; url = urls(metadata))
                 throw(ArgumentError(msg))
             end
 
-            cmd = `wget --http-user=$(username) --http-passwd=$(password) --directory-prefix=$dir $fileurl`
-            run(cmd)
+            Downloads.download(fileurl, filepath; downloader, progress=download_progress)
         end
     end
 
+    # Remove the .netrc file after downloading to avoid storing the credentials
+    remove_netrc!(dir)
+
     return nothing
 end
+
+# ECCO downloader 
+function ECCO_downloader(username, password, dir)
```

Comment:
What is specific about ECCO? This function looks general

Reply:
the netrc file writes down ECCO specific information (the machine for example). We could generalize it easily if we think we might need this somewhere else

Reply:
I have generalized the function to accept a "machine" argument